### ISSUE TITLE
[Windows] AutomationId does not work for ContentView, Layouts and controls that inherit them

### ISSUE THREAD

**Issue by GeorgeDarakchiev** [dotnet/maui#4715]
### Description

Controls like Border or ContentView that have AutomationId are not visible in the AccessibilityTree.
 Therefore they are not located by any UI Automation framework like Appium.

### Steps to Reproduce

1. Run the project from here: [https://github.com/GeorgeDarakchiev/ReproProjects/tree/main/AutomationIdRepro/AutomationIdRepro](https://github.com/GeorgeDarakchiev/ReproProjects/tree/main/AutomationIdRepro/AutomationIdRepro) 
2. Use any accessibility tool of your liking (like UI Verify) in order to inspect the tree. 
3. Notice that the Border and the ContentView from the project are not present in the tree.
4.

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

All

### Did you find any workaround?

As a workaround SemanticProperties.Description can be set, but this leads to issues on iOS.

**Comment by VincentBu** [comment]
verified repro with VS 17.2.0 Preview 2.0 [32301.489.main] in Windows
[4715.zip](https://github.com/dotnet/maui/files/8174088/4715.zip)

**Comment by PureWeen** [comment]
If you use `AutomationProperties.IsInAccessibleTree="true"` does that fix?

**Comment by GeorgeDarakchiev** [comment]
I have just tested it with RC1 and setting `AutomationProperties.IsInAccessibleTree="true"` does not fix the issue.

Even if did, it is still a workaround. I understand that this may be WinUI specific, but I am looking at it from a MAUI point of view. As a cross-platform solution I am expecting to be able to set AutomationId to a ContentView or Border controls and see these controls in all platforms automation trees (with the correct AutomationId).

It doesn't make sense to me that I should set additional property in order to show them in Windows as I can see them in all other platforms.

**Comment by omghb** [comment]
We run into the same issue when we started to support Windows additionally to Android and iOS for our Appium UI tests. The automation tree looks very different on Windows than on Android and iOS. This makes it difficult to re-use the tests.

We try to overcome this issue with following workaround:

```C#
public static class AppBuilderExtensions
{
    public static MauiAppBuilder UseWindowsAutomationTreeFix(this MauiAppBuilder builder)
    {
#if WINDOWS
        Microsoft.Maui.Handlers.ViewHandler.ViewMapper.AppendToMapping(nameof(IView.AutomationId), MapAutomationId);
#endif
        return builder;
    }

#if WINDOWS
    private static void MapAutomationId(IViewHandler handler, IView view)
    {
        // Workaround for https://github.com/dotnet/maui/issues/4715; Layouts are not exposed in AutomationTree
        if (view is not Microsoft.Maui.ILayout || string.IsNullOrEmpty(view.AutomationId)) return;
        var platformView = (Microsoft.UI.Xaml.FrameworkElement?)handler.PlatformView;
        Microsoft.UI.Xaml.Automation.AutomationProperties.SetName(platformView, view.AutomationId);
    }
#endif
}

```

**Comment by jbe2277** [comment]
This issue still exists in MAUI 9.

**Comment by ederbond** [comment]
Still present on MAUI 10

**Comment by ederbond** [comment]
@davidortinau Any information about 4yo bug that's preventing us to build reliable UI Tests for MAUI on Windows?

**Comment by Redth** [comment]
This is similar https://github.com/dotnet/maui/pull/35562/ but doesn't fully cover the ask here (ContentView wouldn't be covered).

I think it's worth building on top of that PR though.

**Comment by ederbond** [comment]
For the God sake yes please @Redth , I've already verified it this week with MAUI 10.0.70 and for the `ContentPage` setting the `AutomationId` it's still buggy on Windows.
It would be good to to check if it's also buggy for `ContentView` too.

BTW week I've spend quite of a time with Opus model trying to workaround it using a `CustomHandler` and as for `ContentViews`, I'm quite sure I've succeded, however as for `ContentPage` Opus 4.* told me that it was implossible to fix it for `ContentPage` because the corresponding WinUI 3 element doesn't have Automation Peers or something like that.

So contact the WinUI3 and make it work for both `ContentView` and `ContentPage`

**Comment by Redth** [comment]
I don't believe the fix in #35562 is in 10.0.70 so I'd expect it to be buggy there.  Once that merged PR does release it should work for ContentPage, ContentView both.  I ultimately only found the logic in that PR is a bit of a behavioural change in that screen readers will pick up and announce those items that have AutomationId set where they didn't before, and arguably shouldn't by default.  My PR #35597 is now updated to address that more minor detail.
